test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -13460,6 +13491,12 @@ def __init__(self, username=None, attrs={}, jobname=None):
             self.username = None
         self.du = None
         self.interactive_handle = None
+        if self.platform == 'cray' or self.platform == 'craysim':
+            m = None
+            if 'Resource_List.select' in attrs:
+                m = re.match('.*vntype=.*', attrs['Resource_List.select'])
+            if 'Resource_List.vntype' not in attrs and m is None:
+                attrs['Resource_List.vntype'] = 'cray_compute'
```

Comment:
If a job is submitted with a select statement, this will cause you trouble.  You can't submit a job with -lvntype and -lselect together.  The way you have it now, if vntype is not in the select, you add -lvntype.  This will only way this will work is if select is not set and PBS is converting the -l resources into a select for the job.